### PR TITLE
TACHYON-198 Move UserConf usage to use TachyonConf

### PR DESCRIPTION
**Pull request by hsaputra**
This PR consist next steps to integrate new configuration work:
-) Remove usage of UserConf
-) Push TachyonConf to Tachyon Hadoop's file system (aka TFS class).
-) Remove UserConf.java from repo.

### CODE REVIEW

File: core/src/main/java/tachyon/conf/TachyonConf.java [hunk 1]
```diff
@@ -58,6 +61,54 @@ public static void assertValidPort(final InetSocketAddress address, TachyonConf
   }
 
   /**
+   * Store the source {@link TachyonConf} object to the target
+   * Hadoop {@link org.apache.hadoop.conf.Configuration} object.
+   *
+   * @param source the {@link TachyonConf} to be stored
+   * @param target the {@link org.apache.hadoop.conf.Configuration} target
+   */
+  public static void storeToHadoopConfiguration(TachyonConf source, Configuration target) {
```
Comment (hsaputra) [inline core/src/main/java/tachyon/conf/TachyonConf.java]: Ok, I will util class for conf to contains these 2 methods

File: core/src/main/java/tachyon/client/AbstractTachyonFS.java [hunk 2]
```diff
@@ -19,12 +19,18 @@
...
 import tachyon.Constants;
 import tachyon.TachyonURI;
-import tachyon.conf.UserConf;
+import tachyon.conf.TachyonConf;
 
 /**
  * Abstract implementation of {@link tachyon.client.TachyonFSCore} APIs.
  */
 abstract class AbstractTachyonFS implements TachyonFSCore {
+  protected final TachyonConf mTachyonConf;
```
Comment (hsaputra) [inline core/src/main/java/tachyon/client/AbstractTachyonFS.java]: +1 for @dcapwell comment, this is just final modifier for instance member.

**Comment by hsaputra** [comment]
@haoyuan updated the PR based on review.

**Comment by haoyuan** [comment]
Thanks!

**Pull request merged**